### PR TITLE
Dynamically render config.json

### PR DESCRIPTION
**Pull request by butonic**
all ocis relevant settings but the list of apps can be set via env vars. We can use the config struct to preconfigure ocis-phoenix from ocis

- [x] update docs in this PR

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -36,9 +36,50 @@ type Asset struct {
...
 }
 
-// Phoenix defines the available phoenic configuration.
+// PhoenixConfig defines the available phoenix configuration for a dynamically rendered config.json.
+type PhoenixConfig struct {
+	Server        string `json:"server,omitempty"`
+	Theme         string `json:"theme,omitempty"`
+	Version       string `json:"version,omitempty"` // TODO what is version used for?
+	OpenIDConnect OIDC   `json:"openIdConnect,omitempty"`
+	// TODO add nilasempty when https://go-review.googlesource.com/c/go/+/205897/ is released
```
Comment (tboerger) [inline pkg/config/config.go:45]: Why would you need that at all?

File: pkg/config/config.go [hunk 2]
```diff
@@ -36,9 +36,50 @@ type Asset struct {
...
+//	"name": "hello",
+//	"path": "http://localhost:9105/hello.js",
+//	  "config": {
+//	    "url": "http://localhost:9105"
+//	  }
+//  }
+type ExternalApp struct {
+	Name string `json:"name,omitempty"`
+	Path string `json:"path,omitempty"`
+	// Config is completely dynamic, because it depents on the extension
```
Comment (refs) [inline pkg/config/config.go]: depends*

File: pkg/flagset/flagset.go [hunk 3]
```diff
@@ -148,5 +148,72 @@ func ServerWithConfig(cfg *config.Config) []cli.Flag {
...
+			 // TODO somehow cast to *StringSlice ... which is a []string
+			 // or find a way to provide a default value
+			&cli.StringSliceFlag{
+				Name:   "web-config-apps",
+				Value:  &cfg.Phoenix.Config.Apps,
+				Usage:  "Version",
+				EnvVar: "PHOENIX_APPS",
+			},
+		*/
+		// TODO EXTERNAL APPS?
```
Comment (refs) [inline pkg/flagset/flagset.go:179]: you need to declare apps:

`apps := cli.StringSlice(cfg.Phoenix.Config.Apps)`

and then these can be passed for to the structure

```go
&cli.StringSliceFlag{
	Name:   "web-config-apps",
	Value:  &apps,
	Usage:  "Version",
	EnvVar: "PHOENIX_APPS",
},
```

File: pkg/service/v0/service.go [hunk 4]
```diff
@@ -56,26 +57,46 @@ func (p Phoenix) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	payload, err = ioutil.ReadFile(p.config.Phoenix.Path)
 
 	if err != nil {
 		p.logger.Error().
 			Err(err).
 			Str("config", p.config.Phoenix.Path).
 			Msg("Failed to read custom config")
 
+	}
+	return payload, err
```
Comment (refs) [inline pkg/service/v0/service.go]: you can write it as `return` since you're using named return values

**Review by refs — APPROVED** [review]
👍 minor changes, and a proposal to initialize apps

File: pkg/flagset/flagset.go (hunk 3 above)
Comment (butonic) [reply pkg/flagset/flagset.go:179]: That will copy the []string ... the code will not pick up the apps

Comment (refs) [reply pkg/flagset/flagset.go:179]: wat? no, that uses type conversion to make cfg.Phoenix.Config.Apps (type []string) into a `cli.StringSlice` but stores the values on it. Unless they're overridden elsewhere, this works just fine, I have a small example here:

```go
package main

import (
	"fmt"
	"os"

	"github.com/urfave/cli"
)

func main() {
	apps := cli.StringSlice([]string{"1", "2"})
	cli.HelpFlag = cli.StringSliceFlag{
		Name:   "web-config-apps",
		Value:  &apps,
		Usage:  "Version",
		EnvVar: "PHOENIX_APPS",
	}

	fmt.Println(cli.HelpFlag.String())
	(&cli.App{}).Run(os.Args)
}
```

File: pkg/service/v0/service.go [hunk 5]
```diff
@@ -56,26 +57,46 @@ func (p Phoenix) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	p.mux.ServeHTTP(w, r)
 }
 
-// Config implements the Service interface.
-func (p Phoenix) Config(w http.ResponseWriter, r *http.Request) {
-	if _, err := os.Stat(p.config.Phoenix.Path); os.IsNotExist(err) {
+func (p Phoenix) getPayload() (payload []byte, err error) {
```
Comment (tboerger) [inline pkg/service/v0/service.go:60]: getPayload? Seriously? We are not writing PHP...

**Pull request merged**